src/content/onlineservice/sopcast_content_handler.cc

```diff
@@ -144,18 +144,18 @@ std::shared_ptr<CdsObject> SopCastContentHandler::getObject(const std::string& g
     // I wish they had a mimetype setting
     //auto mappings = config->getDictionaryOption(CFG_IMPORT_MAPPINGS_EXTENSION_TO_MIMETYPE_LIST);
     //std::string mt = getValueOrDefault(mappings, temp);
-    std::string mt;
-    // map was empty, we have to do construct the mimetype ourselves
-    if (temp == "wmv")
-        mt = "video/sopcast-x-ms-wmv";
-    else if (temp == "mp3")
-        mt = "audio/sopcast-mpeg";
-    else if (temp == "wma")
-        mt = "audio/sopcast-x-ms-wma";
-    else {
+    std::string mt = [temp]() {
+        // map was empty, we have to do construct the mimetype ourselves
+        if (temp == "wmv")
+            return "video/sopcast-x-ms-wmv";
+        if (temp == "mp3")
+            return "audio/sopcast-mpeg";
+        if (temp == "wma")
+            return "audio/sopcast-x-ms-wma";
+        return "application/sopcast-stream";
+    }();
+    if (mt == "application/sopcast-stream")
         log_warning("Could not determine mimetype for SopCast channel (stream_type: {})", temp.c_str());
```

Comment:
Why don't you move that warning into the lambda?

Reply:
Good catch.